### PR TITLE
[Mellanox]Upload the SDK makefile modification

### PR DESCRIPTION
**Pull request by stephenxs**
**- What I did**
Adjust the SDK makefiles so that it reflects the dependencies among libraries in SDK 4.3.2104.
This is a supplement of PR [[Mellanox]Update SDK(4.3.2104), SAI-Implementation(1.15) and firmware 2162.](https://github.com/Azure/sonic-buildimage/pull/3538) It doesn't impact the sonic-mellanox.bin image but makes the future SDK-integration easier.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by stephenxs** [comment]
retest this, please

File: platform/mellanox/sdk.mk [hunk 1]
```diff
@@ -15,17 +15,18 @@ endif
...
-MLNX_SDK_RDEBS += $(APPLIBS) $(IPROUTE2_MLNX) $(SX_COMPLIB) \
-		  $(SX_EXAMPLES) $(SX_GEN_UTILS) $(SX_SCEW) $(SXD_LIBS) $(WJH_LIBS)
+MLNX_SDK_RDEBS += $(APPLIBS) $(IPROUTE2_MLNX) $(SX_COMPLIB) $(SX_EXAMPLES) \
+                  $(SX_GEN_UTILS) $(SX_SCEW) $(SXD_LIBS) $(WJH_LIBS) $(SX_ACL_HELPER)
 
-MLNX_SDK_DEBS += $(APPLIBS_DEV) $(IPROUTE2_MLNX_DEV) \
-		 $(SX_COMPLIB_DEV) $(SX_COMPLIB_DEV_STATIC) $(SX_EXAMPLES_DEV) \
-		 $(SX_GEN_UTILS_DEV) $(SX_SCEW_DEV) $(SX_SCEW_DEV_STATIC) \
-		 $(SXD_LIBS_DEV) $(SXD_LIBS_DEV_STATIC) $(WJH_LIBS_DEV)
+
```
Comment (qiluo-msft) [inline platform/mellanox/sdk.mk]: remove extra empty line

File: platform/mellanox/sdk.mk [hunk 2]
```diff
@@ -15,17 +15,18 @@ endif
...
+MLNX_SDK_DEBS += $(APPLIBS_DEV) $(IPROUTE2_MLNX_DEV) $(SX_COMPLIB_DEV) \
+                 $(SX_COMPLIB_DEV_STATIC) $(SX_EXAMPLES_DEV) $(SX_GEN_UTILS_DEV) \
+                 $(SX_SCEW_DEV) $(SX_SCEW_DEV_STATIC) $(SXD_LIBS_DEV)\
+                 $(SXD_LIBS_DEV_STATIC) $(WJH_LIBS_DEV) $(SX_ACL_HELPER_DEV)
```
Comment (qiluo-msft) [inline platform/mellanox/sdk.mk]: >$(SX_ACL_HELPER_DEV [](start = 56, length = 19)

If you can add your new item at a new line and keep old lines, that would be better.

File: platform/mellanox/sdk-src/sx-acl-helper/Makefile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+DERIVED_TARGETS = sx-acl-helper-dev_1.mlnx.$(MLNX_SDK_DEB_VERSION)_amd64.deb \
+				  sx-acl-helper-dev-static_1.mlnx.$(MLNX_SDK_DEB_VERSION)_amd64.deb \
+				  sx-acl-helper-dbgsym_1.mlnx.$(MLNX_SDK_DEB_VERSION)_amd64.deb
+PACKAGE_NAME = sx_acl_helper
+
+$(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
+	# get sources
+	rm -rf $(PACKAGE_NAME)-$(MLNX_SDK_VERSION)-$(MLNX_SDK_ISSU_VERSION)
+
+	wget -c $(MLNX_SDK_SOURCE_BASE_URL)/$(PACKAGE_NAME)-$(MLNX_SDK_VERSION)-$(MLNX_SDK_ISSU_VERSION).tar.gz -O - | tar -xz
```
Comment (qiluo-msft) [inline platform/mellanox/sdk-src/sx-acl-helper/Makefile]: >wget -c [](start = 1, length = 7)

maybe curl is easier than wget to get stdout.

File: platform/mellanox/sdk-src/sx-acl-helper/Makefile [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+	if [ -f autogen.sh ]; then
+		./autogen.sh
+	fi
+
+	debuild -b -us -uc -j$(SONIC_CONFIG_MAKE_JOBS)
+
+	popd
+
+	mv $(DERIVED_TARGETS) $* $(DEST)/
```
Comment (qiluo-msft) [inline platform/mellanox/sdk-src/sx-acl-helper/Makefile]: >$* [](start = 23, length = 2)

Could you explain `$*`?

File: platform/mellanox/sdk.mk (hunk 1 above)
Comment (stephenxs) [reply platform/mellanox/sdk.mk]: > remove extra empty line

will remove.

File: platform/mellanox/sdk.mk (hunk 2 above)
Comment (stephenxs) [reply platform/mellanox/sdk.mk]: I just found that the length of the old lines is uneven and intended to beautify them.

File: platform/mellanox/sdk-src/sx-acl-helper/Makefile (hunk 4 above)
Comment (stephenxs) [reply platform/mellanox/sdk-src/sx-acl-helper/Makefile]: Hi, it stands for the default deb package. The sx-acl-helper_1.mlnx.4.3.2104_amd64.deb, in this case.

File: platform/mellanox/sdk-src/sx-acl-helper/Makefile (hunk 3 above)
Comment (stephenxs) [reply platform/mellanox/sdk-src/sx-acl-helper/Makefile]: Hi, since wget works well here, let's keep it as it is.

**Comment by stephenxs** [comment]
retest vs, please

**Comment by stephenxs** [comment]
retest this, please

**Pull request merged**